### ISSUE TITLE
'locale' missing in Docker image?

### ISSUE THREAD

**Issue by saketkc** [bioconda/bioconda-utils#70]

**Comment by saketkc** [comment]
I am trying to build a recipe in `'python3.5` that uses `python-click`.  `click` runs a [unicode check](https://github.com/pallets/click/blob/master/click/_unicodefun.py#L65) for `PY3` envs, but this seems to return a `FileNotFoundError` on [travis](https://travis-ci.org/saketkc/bioconda-recipes/jobs/203384804#L1053)

I am not familiar with Docker, but would setting `ENV LANG en_US.UTF-8` in the Dockerfile, fix the issue?

**Comment by saketkc** [comment]
Actually, I am not sure why the `locale -a` command should fail on a linux image.

**Comment by johanneskoester** [comment]
Does it maybe fail in the BusyBox test @bgruening?

**Comment by bgruening** [comment]
@saketkc thats true, local is not part of the busybox Docker image. Should we get locale into conda-forge or should we add it into the busybox image? I'm more in favor of the conda-forge solution as this seems to be a very special case.

**Comment by saketkc** [comment]
@bgruening I am unaware of the architecture, but happy to help, in my capacity.

**Comment by saketkc** [comment]
@bgruening Any luck on this one?

**Comment by emarinier** [comment]
Hello,

I believe I'm having a similar issue with Click in my pull request:

https://github.com/bioconda/bioconda-recipes/pull/11206

The [first click error](https://circleci.com/gh/bioconda/bioconda-recipes/28450?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) I received was:

> Additional information: on this system no suitable UTF-8 locales were discovered.  This most likely requires resolving by reconfiguring the locale system.

I tried changing my meta.yml file to specifically include a locale, [similar to this project](https://github.com/bioconda/bioconda-recipes/blob/master/recipes/moca/meta.yaml):

```
test:
  commands:
    - LANG=C.UTF-8 quasitools --help
```

However, this gave [another error](https://circleci.com/gh/bioconda/bioconda-recipes/28459?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) and some additional information:

> This system lists a couple of UTF-8 supporting locales that you can pick from.  The following suitable locales were discovered: en_US.utf8
>
> Click discovered that you exported a UTF-8 locale but the locale system could not pick up from it because it does not exist.  The exported locale is "C.UTF-8" but it is not supported

Finally, I tried changing my meta.yml file to use `en_US.utf8` instead of `C.UTF-8`:

```
test:
  commands:
    - LANG=en_US.utf8 quasitools --help
```

However, this gave [another error](https://circleci.com/gh/bioconda/bioconda-recipes/28471?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link):

> Additional information: on this system no suitable UTF-8 locales were discovered.  This most likely requires resolving by reconfiguring the locale system.
>
> Click discovered that you exported a UTF-8 locale the locale system could not pick up from it because it does not exist.  The exported locale is "en_US.utf8" but it is not supported

I'm uncertain of how to proceed. Does anyone have any suggestions for how to resolve this problem?

**Comment by emarinier** [comment]
As a follow up, we were not able to determine a solution.

We found that the pull request eventually passed all checks successfully with seemingly no changes from the initial attempt. Likewise, after merging the (successful) pull request, Bioconda failed continuous integration and deployment. We had to run the same workflow multiple times until it passed, with no changes in code between the runs.

My hypothesis is that there are some build environments that have locales and some that don't. However, I don't have an explanation for why that might be the case.
